Draft: Add assets and code for missing infested blocks

pl-luk opened the pull request:
This PR adds assets for blocks that are currently not being infested (As villages and player structures currently look a little bit ugly when infested).

_Note_: In order to get blocks with blockstates (i.e. stairs) to work the `Cursor` and `InfestationTableEntry` code had to be altered slightly.

pl-luk committed:
Add code to support infecing more complex blocks

pl-luk committed:
Add wood stairs

pl-luk committed:
Add cobblestone stairs

pl-luk committed:
Fix BlockState not copied on BlockInfestationEntries

pl-luk committed:
Add stone stairs

pl-luk committed:
Change order of setting table entries to be more inline with original behaviour

pl-luk committed:
Add mossy cobblestone stairs

pl-luk committed:
Fix Blockstate not copied on purification for BlockInfestationTableEntry

Mikeatron-User commented:
Wow, this is great, I will review this and see about pulling this into the main branch.

Mikeatron-User commented:
This PR completely put cursor pathing out of wack, but I'm working to fix it.

pl-luk commented:
Yes the cursor pathing is completely messed up, it seemed like a good idea at the time but after a little bit of playtesting I ran into a lot of issues... I have a semi working solution which consists of saving allowed non-full block blockstates in a list and checking if they can be replaced. I am not really understanding how it failed though.

Mikeatron-User committed:
Refactor of Block Infestation System

Made separate infestation tables for different types of infectable blocks.

Unified the idea of what a solid is.

Moved the initialization of the tables into the BlockInfestationHelper class.

Refactor of Cursor Pathing

Refactor of Infestation Process

Refactor of Purity Process

Mikeatron-User commented:
Finished Feature and has been merged into the 1.20 branch. Expect an update to the mod soon.

Mikeatron-User commented:
If you have a discord @ Me and ill give you the contributor role @pl-luk
